app/src/main/java/de/xikolo/storages/RecentCoursesStorage.kt

```diff
@@ -0,0 +1,55 @@
+package de.xikolo.storages
+
+import android.content.Context
+import androidx.preference.PreferenceManager
+import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
+import de.xikolo.App
+import de.xikolo.storages.base.BaseStorage
+import de.xikolo.storages.base.SharedPreferenceLiveData
+
+class RecentCoursesStorage : BaseStorage(PREF_RECENT_COURSES, Context.MODE_PRIVATE) {
+
+    val recentCourses: LinkedHashSet<Pair<String, String>>?
```

Comment:
It would be nice to introduce a `typealias` for `Pair<String, String>` or something similar to have a better readable API. This way you could avoid obscure calls like `course.first` and call something like `course.id` instead.